cmd/nvidia-ctk-installer/container/runtime/containerd/config_test.go

```diff
@@ -0,0 +1,978 @@
+				initialConfig := `version = 1
+
+[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      default_runtime_name = "runc"
+
+      [plugins.cri.containerd.runtimes]
+        [plugins.cri.containerd.runtimes.runc]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.runc.options]
+            Runtime = "/usr/bin/runc"
+`
+				require.NoError(t, os.WriteFile(configPath, []byte(initialConfig), 0600))
+				return nil
+			},
+			assertSetupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+				require.FileExists(t, configPath)
+
+				actual, err := os.ReadFile(configPath)
+				require.NoError(t, err)
+
+				expected := `version = 1
+
+[plugins]
+
+  [plugins.cri]
+
+    [plugins.cri.containerd]
+      default_runtime_name = "runc"
+      enable_cdi = true
+
+      [plugins.cri.containerd.runtimes]
+
+        [plugins.cri.containerd.runtimes.nvidia]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime"
+            Runtime = "/usr/bin/nvidia-container-runtime"
+
+        [plugins.cri.containerd.runtimes.nvidia-cdi]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia-cdi.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime.cdi"
+            Runtime = "/usr/bin/nvidia-container-runtime.cdi"
+
+        [plugins.cri.containerd.runtimes.nvidia-legacy]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia-legacy.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime.legacy"
+            Runtime = "/usr/bin/nvidia-container-runtime.legacy"
+
+        [plugins.cri.containerd.runtimes.runc]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.runc.options]
+            Runtime = "/usr/bin/runc"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+				require.FileExists(t, configPath)
+
+				actual, err := os.ReadFile(configPath)
+				require.NoError(t, err)
+
+				expected := `version = 1
+
+[plugins]
+
+  [plugins.cri]
+
+    [plugins.cri.containerd]
+      default_runtime_name = "runc"
+      enable_cdi = true
+
+      [plugins.cri.containerd.runtimes]
+
+        [plugins.cri.containerd.runtimes.runc]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.runc.options]
+            Runtime = "/usr/bin/runc"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+		},
+		{
+			description:     "v1 legacy: existing config with default runtime",
```

Comment:
One of the things that we should test in this case is that the OPTIONS from the default runtime are inherrited. The test as implemented does not do this.